angular/openmrs-rest/openmrs-rest.js

```diff
@@ -67,6 +67,15 @@ function httpProviderConfig($httpProvider){
 	$httpProvider.defaults.headers.common['Disable-WWW-Authenticate'] = 'true';
 }
 
+function getBasePath(pathname) {
+    if (pathname.indexOf('/coreapps/') !== -1) {
+        return pathname.substring(0, pathname.indexOf('/coreapps/'));
+    } else {
+        return pathname.substring(0, pathname.indexOf('/owa/'));
+    }
+
+}
```

Comment:
I guess it's good enough for now, but it would be nice to be actually able to override restBasePath in coreapps or any other module by calling some method on openmrsRest before using it.